ablate/common/optim.py

```
@@ -145,13 +156,24 @@ def build_optimizers(model, muon_lr=3e-4, adam_lr=3e-4, wd=0.1, momentum=0.95, n
     # ~0.5 over 2000 steps, so it needs its own group and its own lr. wd=0: decay on an exponent
     # would just re-impose the lr/wd equilibrium this axis exists to escape.
     _is_as = lambda n: "radial_theta" in n or "xsa_alpha" in n
+    _is_typed_read = lambda n: (
+        "_res_type_controller" in n
+        or "_res_type_bias" in n
+        or "typed_attn_res_" in n
+    )
```

Ruff (0.16.0)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `_is_typed_read` as a `def`

(E731)